lib/vehicle_factory.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+
+class VehicleFactory
+  attr_reader :vehicles
+
+  def initialize
+    @vehicles = []
+  end
+
+  def create_vehicles(database)
```

Comment:
Your `create_vehicles` method is doing a lot. It's determining the data source, using that determination to format data appropriately, and then, finally, it creates vehicles. I'd break it down more like this:
```ruby
def create_vehicles(database)
    vehicle_data = format_vehicle_data(database)
    vehicle_data.each do |vehicle|
      @vehicles << Vehicle.new(vehicle)
    end
  end

  def format_vehicle_data(database)
    if database.first.keys.include?(:scofflaw_indicator)
      format_ny_vehicles(database)
    else
      format_wa_vehicles(database)
    end
  end

  def format_wa_vehicles(database)
    database.map do |vehicle|
      {
        vin: vehicle[:vin_1_10],
        year: vehicle[:model_year],
        make: vehicle[:make],
        model: vehicle[:model],
        engine: :ev
      }
    end
  end

  def format_ny_vehicles(database)
    vehicle_database = database.find_all do |record|
      record[:record_type] == "VEH"
    end
    vehicle_database.map do |vehicle|
      engine = :ice unless vehicle[:fuel_type] != "GAS"
      {
        vin: vehicle[:vin],
        year: vehicle[:model_year],
        make: vehicle[:make],
        model: vehicle[:body_type],
        engine: engine
      }
    end
  end
```
You might even consider initializing with the `database` so that you don't need to pass it around as an argument so much in my proposed refactor above.